### ISSUE TITLE
AVT 3 - React Toolbar needs improvements for VO  keyboard navigation on iOS and macOS

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#3381]
1. Start VO on macOS
2. Tab to the Toolbar, both Search and Filter icons have focus. VO announces, “Search. You are currently on a text field. To enter this field, type.” (see screenshot)
<img width="720" alt="Screen Shot 2019-07-11 at 5 07 47 PM" src="https://user-images.githubusercontent.com/21676914/61142024-6bab1500-a494-11e9-8fbc-e72eef41bfa8.png"> **Expected Result:** The focus should only be on the Search icon and advise the user that it is a button.  The user should not be able to type anything yet. If the user does type a word it is saved in the search text field but the user can not use the Space or Enter key to submit the search and VO is silent.  

3. The only way to exit the search field is to Tab which takes the user to the Search button icon and VO announces that it is a button. (see screenshot)
<img width="772" alt="Screen Shot 2019-07-11 at 5 08 23 PM" src="https://user-images.githubusercontent.com/21676914/61142063-7c5b8b00-a494-11e9-9211-6cd70ee8dff4.png">**Expected Result:** The purpose of the button should be announced. 

4. Press the Space key to open the search and VO announces, “Search. You are currently on a text field. To enter this field, type.” Note: The text that was entered in step 2 is in the text field and the X icon that displays cannot get focus to clear the text field.
**Expected Result:** Text should not be present yet and the X icon should clear the text field.

5. Press Tab again and the focus moves back to the Search button and the Search field remains open. 
**Expected Result:** After the search, Tab should move to the X to Clear the search and then Tab again should move to the Filter button.

5. Press Tab again and the focus moves to the filter icon after the search icon and VO announces, “Menu pop-up button.  You are currently on a pop-up button. To select this pop-up button press Control-Option-Space.”
**Expected Result:** The purpose of the pop-up menu should be announced. 

6. Press the Space key and the Filter selections display. VO announces that you are on a menu that displays a list of  4 items and to move press Control-Option-Space. Moving to the first item on the list it is announced as 1 of 5, 2 of 5, 3 of 5, and 4 of 5. 
**Expected Result:** VO should announce that this is a list of 3 items… and moving to each item VO should announce the option 1 of 3 and the state, option 2 of 3 and the state, and option 3 of 3 and the state.

7. Press Esc Close the menu and then press Tab and VO announces, “Menu pop-up button.  You are currently on a pop-up button. To select this pop-up button press Control-Option-Space.”
**Expected Result:** The purpose of the pop-up menu should be announced.

8. Press the Space key and the Menu selections display. VO announces that you are on a menu that displays a list of  5 items and to move press Control-Option-Space. Moving to the first item on the list the Refresh Table text is announced as 1 of 6 and the remaining items are announced 2 of 6, 3 of 6, etc. 
**Expected Result:** VO should announce that this is a list of 3 items… and moving to each item should be Refresh table 1 of 3, Short Radio button 2 of 3 and the state, Tall radio button 3 of 3 and the state. Also, note that the horizontal rule does not need to have focus and it should not be counted as an item. 

9. Press Esc to close the pop-up menu.

WCAG 2.0 Requirements:
[Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)
[Meaningful Sequence: Understanding SC 1.3.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-sequence.html)

Similar issues on iOS with VO when using gestures. iOS has addition focus issues when navigating in and out of pop-up menus.
